lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4270,6 +4345,8 @@ ParseResult FIRCircuitParser::parseClass(CircuitOp circuit, unsigned indent) {
   deferredModules.emplace_back(
       DeferredModuleToParse{classOp, portLocs, getLexer().getCursor(), indent});
 
+  // Stash the class name -> op in the constants, so we can resolve Inst types.
```

Comment:
Can type aliases refer to this, are there ordering issues? IIRC type aliases themselves have some requirements about ordering (probably def-before-use? not sure), just wondering.

Reply:
Re type aliases, this parses (albeit with a warning that the alias is a non-base-type):
```
FIRRTL version 3.2.0
circuit Top:
  class MyClass:

  type MyAlias = Inst<MyClass>

  module Top:
    output port : MyAlias
    object myclass of MyClass
    propassign port, myclass

```
to
```
module {
  firrtl.circuit "Top" {
    firrtl.class private @MyClass() {
    }
    firrtl.module @Top(out %port: !firrtl.class<@MyClass()>) attributes {convention = #firrtl<convention scalarized>} {
      %0 = firrtl.object @MyClass()
      firrtl.propassign %port, %0 : !firrtl.class<@MyClass()>
    }
  }
}
```

(You cannot use a type alias to instantiate the class).

Re ordering: yes, there are ordering concerns, but it is mainly an accident of how our parser works. Because our parser makes two passes over the input, once for parsing the interfaces of classes/modules, and a second time to parse the body, a class has to be defined before used if it is referenced by a port's type.